### DISCUSSION TITLE
shrpi daemon "Error: Received HTTP status 400"

### DISCUSSION THREAD

**Discussion by psm12** [hatlabs/discussions#62] [Q&A]
I'm just tinkering with the 'shrpi sleep' command . It works as I expected using a number (seconds) argument but if I use a time format (00:00 or 00:00:00), I get the response:
"Error: Received HTTP status 400"

(Edit: maybe because I don't have a battery in the sh rtc?)

Also, thinking about the 'shrpi shutdown' command, why does the response to the Linux 'shutdown' command not trigger the watchdog to reboot the pi?

I'm also wondering what the difference is between 'shrpi sleep' and shrpi shutdown'. Maybe I'm overthinking?

Paul

**Comment by mairas** [comment]
Just checked and noticed that the help string of `shrpi sleep --help` is wrong: you need to provide a full datetime string as input. Sorry about that!

For example:

`shrpi sleep '2023-08-17T23:00:00'`

or

`shrpi sleep '2023-08-17 23:00'`

It seems like `shutdown now` on Raspberry Pi doesn't actually halt the system but leaves it alive as a 'zombie'. Out of habit, I have always used `shutdown -h now` myself, and that does halt the system. `shrpi shutdown` invokes `poweroff` which should be identical to `shutdown -h now`, but it also sends a shutdown request to the SH-RPi board, causing it to cut the power if the device doesn't shutdown in time.

`shrpi sleep` is a completely different beast. It leaves the power cut off and the MCU waiting for the real-time clock alarm. The internal state machine showing the different states can be seen here: https://github.com/hatlabs/SH-RPi-firmware#state-machine

**Reply by psm12** [reply]
Thanks Matti,

I tried the syntax and it works but I think there may be a problem with the logic. The wake-up seems to be a full hour later than expected. ( local time (dst) =ut+1 ) but the sh seems to wake up the pi at the time set + 1 additional hour.

Also the manual wake-up (discussed in the previous thread)is very desirable as it takes >15 minutes to discharge the supercaps to allow the sh to be power cycled when in sleep mode. 

Paul

**Reply by mairas** [reply]
shrpi at the moment assumes dates are in UTC. This is not intuitive; I'll fix it.

**Reply by mairas** [reply]
I've now released a new version of SH-RPi-daemon that should fix the date parsing issues. You can now give the sleep time as pure time, as a datetime or as relative time such as "in 10 minutes". Also the numeric pure seconds argument is still supported.
